### ISSUE TITLE
Investigate using an ORM to simplify queries

### ISSUE THREAD

**Issue by tombola** [vitalseeds/vs-data-api#4]
An ORM would save us creating SQL queries directly.

SQLalchemy can query odbc directly (via pypyodbc?).

SQLmodel is a wrapper around SQLalchemy that might simplify things.

**Comment by tombola** [comment]
SQL Alchemy _can_ query a database using ODBC, but this is not practical for our use case.

Firstly my development machine is a mac and the FM I am developing against is also an older version (FM18). As far as I can understand it this means that the the filemaker oDBC drivers are built on IODBC, rather than UnixODBC (ODBC is a microsoft standard, these are the two solutions to implement in on *NIX). The normal python ODBC library, pyodbc, will not seem to work against FM for this reason, I believe because it uses UnixODBC by default. A quick look at building the package with IODBC instead was a no-go, too complex.

This is a problem because SQL Alchemy uses pyodbc (so can't talk to FileMaker).

With some quick testing actually think could get around all that by [tweaking](https://github.com/sqlalchemy/sqlalchemy/issues/2795#issuecomment-441920304) SQL Alchemy to use pypyodbc instead. That seems a bit fragile though.

Even with SQL Alchemy using pypyodbc, it looks like would also need to create a SA 'dialect' to compensate for FM's SQL quirks. Tried the MSSQL dialect over (py)pyodbc, did seem to actually connect, but Filemaker didn't like the SQL. Creating the dialect is a step too far - it seems likely that even if I got it together, FM would simply lack some of the required statements/concepts.

Remaining chance would be to try the mysql+pyodbc SA dialect+connector, just in case there is overlap there.

Basically it looks like if I want an ORM I might have to implement a very limited one myself, and fallback to SQL.
